tools/decompile.py

```diff
@@ -1,338 +1,324 @@
+        self.relpath: Path = function_path.relative_to(func_root)
+        self.src_path: Path = self._infer_src_path()
+        if args.version == "saturn":
+            self.include_asm: str = (
+                f'INCLUDE_ASM("{self.abspath.relative_to(self.root).parent}", {self.relpath.stem}, {self.name});'
+            )
+        else:
+            self.include_asm: str = (
+                f'INCLUDE_ASM("{self.abspath.relative_to(self.asm_dir).parent}", {self.relpath.stem});'
+            )
+        self.asm_code: str = self.abspath.read_text()
+        self.c_code: str = ""
+        self._context: str = ""
+
+    @property
+    def overlay(self) -> str:
+        """Attempts to infer the overlay from the asm file path or returns None if one is not found.
+        The '_psp' extension will be included if the version is psp"""
+        nonmatchings_dir = (
+            "f_nonmat"
+            if args.version == "saturn"
+            else "psp" if "psp" in args.version else "nonmatchings"
+        )
+        if not self._overlay:
+            nonmatching_index = next(
+                (
+                    i
+                    for i, part in enumerate(self.relpath.parts)
+                    if part == nonmatchings_dir
+                ),
+                None,
+            )
+            self._overlay = (
+                self.relpath.parts[nonmatching_index - 1] if nonmatching_index else None
+            )
+        return self._overlay
+
+    @property
+    def asm_differ_command(self) -> str:
+        """Constructs the command string the user can enter to run asm-differ against the overlay this function resides in"""
+        return f"{Path(sys.executable).name} {self.root.joinpath("tools/asm-differ/diff.py").relative_to(Path.cwd())} -mwo --overlay {self.overlay} {self.name}"
+
+    @property
+    def context(self):
+        """Leverages m2ctx to create context text"""
+        if not self._context:
+            m2ctx.root_dir = self.root
+            m2ctx.VERSION_DEF = [f"-D_internal_version_{self.version}"]
+            self._context = m2ctx.import_c_file(self.src_path)
+        return self._context
+
+    def decompile(self) -> str:
+        """Leverages m2c to decompile the function and stores it so that decompilation only needs to run once"""
+        if self.version != "saturn" and not self.c_code:
+            self.c_code = self._guess_unknown_type(self._run_m2c())
+        return self.c_code
+
+    def upload_to_decompme(self) -> None:
+        """Uploads the decompiled code to decomp.me using the extracted context and SotN decomp.me presets"""
+        print("Loading up the catapult...")
+        # No current expectation of these changing, so hardcoding instead of retrieving them from the API programmatically is probably fine
+        # The preset id's can be found at https://www.decomp.me/api/compilers, searching the page for Castlevania: Symphony of the Night
+        presets = {"us": 14, "pspeu": 132, "saturn": 21}
```

Comment:
```suggestion
        presets = {"us": 14, "hd": 14, "pspeu": 132, "saturn": 21}
```
If we want to go ahead with using a map of presets, we need to support the HD version

Reply:
This was my oversight.  I intended to shift this to go by platform, rather than version.

Reply:
It also ended up being cleaner to go with your original suggestion, so I went with that.